### PR TITLE
Refactor telegram runner

### PR DESCRIPTION
**Pull request by bavaz1**

### CODE REVIEW

File: src/utils/utils.ts [hunk 1]
```diff
@@ -35,340 +31,22 @@ const getErrorResult = (error: any): ErrorResult => {
   };
 };
 
-const logAxiosResponse = (res: AxiosResponse<any>): AxiosResponse<any> => {
+const logAxiosResponse = (res: AxiosResponse<any>) => {
+  const data = JSON.stringify(res.data);
   logger.verbose(
```
Comment (bessbd) [inline src/utils/utils.ts:36]: Please ensure this is the lowest possible log level.

File: src/api/service.ts [hunk 2]
```diff
@@ -14,26 +15,38 @@ import {
...
           }
-
-          return await kickUser(
-            +platformGuildId,
-            +platformUserId,
-            "have not fulfilled the requirements, disconnected your Telegram account or just left it."
-          );
+          if (action === "REMOVE") {
+            const guild = await getGuild(platformGuildId);
+            const groupName = await getGroupName(+platformGuildId);
```
Comment (bessbd) [inline src/api/service.ts:41]: Can we bundle these in a `Promise.all`?

File: src/service/commands.ts [hunk 3]
```diff
@@ -24,104 +13,59 @@ const helpCommand = async (ctx: Ctx): Promise<void> => {
...
-  const platformUserId = message.from.id;
 
   const currTime = new Date().getTime();
 
   try {
-    const sender = await Bot.client.getChatMember(
-      platformUserId,
-      platformUserId
-    );
+    await Bot.client.getMe();
```
Comment (bessbd) [inline src/service/commands.ts:58]:
```suggestion

```

File: src/service/commands.ts [hunk 4]
```diff
@@ -24,104 +13,59 @@ const helpCommand = async (ctx: Ctx): Promise<void> => {
...
 
     await ctx.replyWithMarkdownV2(
       markdownEscape(
-        `Pong. @${sender.user.username} latency is ${
+        `Pong. The message latency is ${
```
Comment (bessbd) [inline src/service/commands.ts:62]: Just a pong maybe?

File: src/service/common.ts [hunk 5]
```diff
@@ -1,18 +1,17 @@
-import dayjs from "dayjs";
-import { isMember } from "../api/actions";
+// import { isMember } from "../api/actions";
```
Comment (bessbd) [inline src/service/common.ts:1]:
```suggestion
```

File: src/service/events.ts [hunk 6]
```diff
@@ -203,63 +87,84 @@ const onUserJoined = async (
...
-            "If this is not the case then the admins did not set up the guild properly."
+      if (!access || access.roles?.length === 0) {
+        const kickMessage =
+          reason ??
+          `You do not have access to this reward. To check the requirements, visit here: ${guild.url}`;
+        await kickUser(groupId, newChatMember.user.id, kickMessage);
+        await Bot.client.sendMessage(
+          invitator.id,
+          `You can not invite ${newChatMemberName} to "${groupTitle}" chat, because it does not have access to this reward.`
         );
```
Comment (bessbd) [inline src/service/events.ts:135]: We should avoid sending the message, since we are not performing the kick.

File: src/utils/logger.ts [hunk 7]
```diff
@@ -1,36 +1,38 @@
...
+const createCustomLogger = () => {
+  const devLogFormat = printf((log) => {
+    let msg = `${log.timestamp} ${log.level}: ${log.message}`;
+    if (log.stack) {
+      msg += log.stack;
+    }
+
+    if (config.nodeEnv === "production") {
+      msg = msg.replaceAll("\n", "\t");
+    }
```
Comment (bessbd) [inline src/utils/logger.ts:15]: Consider introducing a log-specific env var instead of switching on environment.

**Review by ejayxyz — APPROVED** [review]
lgtm.

**Pull request merged**